src/main/java/slimeknights/tconstruct/library/TinkerRegistry.java

```diff
@@ -385,7 +385,9 @@ public static void registerToolCrafting(ToolCore tool) {
 
   /** Adds a tool to the Crafting UI of the Tool Station */
   public static void registerToolStationCrafting(ToolCore tool) {
-    toolStationCrafting.add(tool);
+    if(new TinkerRegisterEvent.ToolStationCraftingRegisterEvent(tool).fire()) {
+      toolStationCrafting.add(tool);
+    }
```

Comment:
Missing the debug trace if registering fails. Literally do exactly what we do on other events.